dataprofiler/profilers/profile_builder.py

```diff
@@ -689,13 +689,18 @@ def tqdm(l):
                     options=structured_options
                 )
                 new_cols.add(col)
-                        
-        # Format the data
-        notification_str = "Finding the Null values in the columns..."
+                
+        # Generate pool and estimate datasize
         pool = None
+        est_data_size = df[:50000].memory_usage(index=False, deep=True).sum()
+        est_data_size = (est_data_size / min(50000, len(df))) * len(df)
+        pool, pool_size = utils.generate_pool(
+            max_pool_size=None, data_size=est_data_size, cols=len(df.columns))
```

Comment:
Basically, sample and then estimate the data_size. It's accurate 2-10% based on what I have seen, which is more than accurate enough to select the number of processes to use.